frontend/packages/dev-console/src/components/import/AdvancedOptions.tsx

```diff
@@ -0,0 +1,90 @@
+import * as React from 'react';
+import { filter } from 'lodash';
+import { scrollToBottom } from '../../utils/scroll-utils';
+import './AdvancedOptions.scss';
+
+type AdvancedOptionProps = {
+  options: {
+    [name: string]: {
+      component: React.ElementType;
+      props?: {};
+    };
+  };
+};
+
+type AdvancedOptionState = {
+  optionsKeys: string[];
+  optionsOrder: string[];
+};
+
+class AdvancedOption extends React.PureComponent<AdvancedOptionProps, AdvancedOptionState> {
+  constructor(props: AdvancedOptionProps) {
+    super(props);
+    this.state = {
```

Comment:
Also you don't need to use constructor anymore for class components, you can directly declare your state like `readonly state`.

Reply:
We should use hooks.
However by setting `state` in the constructor, you benefit from the declaration of `state` in `React.Component` instead of having to redefine the `state` type again.